internal/plugincommon/consensus/threshold.go

```diff
@@ -0,0 +1,51 @@
+package consensus
+
+import (
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+)
+
+// Threshold is the minimum number of observations required for a given chain.
+type Threshold uint
+
+// MultiThreshold is an interface that maps a chain selector to a threshold.
+type MultiThreshold interface {
+	// Get the minimum number of observations required for a given chain.
+	Get(chain cciptypes.ChainSelector) (Threshold, bool)
```

Comment:
In my case it's not always a ChainSelector. I use a map by Account and need thresholds for them as well. Can you maybe make it more generic?

Reply:
Done, now allows using a generic key.